### PR TITLE
User per-role cache key for the organization_profile_viewset

### PR DESCRIPTION
**Pull request by FrankApiyo**
### Changes / Features implemented
- Invalidate cache when organization permissions are updated
- User per-role cache key for the organization_profile_viewset

### Steps taken to verify this change does what is intended
- Updated tests
### Side effects of implementing this change
N/A

**Before submitting this PR for review, please make sure you have:**

  - [x] Included tests
  - [ ] Updated documentation

Closes https://github.com/onaio/onadata/issues/2672

### CODE REVIEW

File: onadata/apps/api/tasks.py [hunk 1]
```diff
@@ -36,6 +38,14 @@
 User = get_user_model()
 
 
+def invalidate_organization_cache(org_username):
```
Comment (kelvin-muchiri) [inline onadata/apps/api/tasks.py:41]: Can we move this to `logger_tools`?

Comment (FrankApiyo) [reply onadata/apps/api/tasks.py]: I've moved it  to `cache_tools` instead

File: onadata/apps/api/viewsets/organization_profile_viewset.py [hunk 2]
```diff
@@ -79,20 +88,21 @@ def create(self, request, *args, **kwargs):
         response = super().create(request, *args, **kwargs)
         organization = response.data
         username = organization.get("org")
-        cache.set(f"{ORG_PROFILE_CACHE}{username}{request.user.username}", organization)
+        organization_profile = OrganizationProfile.objects.get(user__username=username)
+        cache_key = get_cache_key(request.user, organization_profile)
+        cache.set(cache_key, organization)
```
[the remark this answers is not in the document]
Comment (kelvin-muchiri) [reply onadata/apps/api/viewsets/organization_profile_viewset.py:93]: @FrankApiyo I've just realized `set_cache_for_org` is in `test_tasks` and not in this file

File: onadata/apps/api/viewsets/organization_profile_viewset.py [hunk 3]
```diff
@@ -115,21 +125,19 @@ def members(self, request, *args, **kwargs):
...
-                organization, context={"request": request}
-            ).data
-            cache.set(f"{ORG_PROFILE_CACHE}{username}{request.user.username}", data)
-        else:
+        if not serializer.is_valid():
             return Response(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+        serializer.save()
+
+        data = OrganizationSerializer(
```
Comment (kelvin-muchiri) [inline onadata/apps/api/viewsets/organization_profile_viewset.py:133]: We should use instead `self.get_serializer_class`

**Review by ukanga — APPROVED** [review]

**Review by kelvin-muchiri — APPROVED** [review]

**Pull request merged**